gen/zap.go

```diff
@@ -86,6 +86,28 @@ func (z *zapGenerator) zapEncoder(g Generator, spec compile.TypeSpec) string {
 	panic(root)
 }
 
+// zapTypedefHasGeneratedMarshaler defines if a typedef will have generated code
+// zapcore.ObjectMarshaler or zapcore.ArrayMarshaler.
+func (z *zapGenerator) zapTypedefHasGeneratedMarshaler(g Generator, spec compile.TypeSpec) bool {
+	switch z.zapEncoder(g, spec) {
+	case "Array", "Object":
+		return true
+	default:
+		return false
+	}
+}
+
+// zapTypedefGenerateMarshaler is similar to zapMarshaler but always converts a typedef
+// to the next value. This should be used when generating the code for the Zap
```

Comment:
you mean ~~next~~ `root` value?

Reply:
In this case it's actually next (`spec.Target`) instead of root. Considering the types in `https://github.com/thriftrw/thriftrw-go/pull/464/files/bbdb77f0f1437cac94ff1619d508168e847cc015#diff-9634c2ea01f5b97b2d07789614e0893c96f71e0b34db720c2c884d2b14fe8ad2`, using root on the example `MyUUID` would cast it to go value for `i128`, which could still cause an unexpected import. Using next makes sure we use the go type for `UUID`.

Reply:
ohh gotcha. next as next in the alias chain.